src/MSBuild/XMake.cs

```diff
@@ -233,14 +233,14 @@ private static void HandleConfigurationException(Exception ex)
 #if FEATURE_APPDOMAIN
         [LoaderOptimization(LoaderOptimization.MultiDomain)]
 #endif
-#pragma warning disable SA1111, SA1009 // Closing parenthesis should be on line of last parameter
-        public static int Main(
-#if !FEATURE_GET_COMMANDLINE
-            string[] args
-#endif
-            )
-#pragma warning restore SA1111, SA1009 // Closing parenthesis should be on line of last parameter
+        public static int Main(string[] args)
         {
+            // When invoked from SDK, insert the command executable path as the first element of the args array.
```

Comment:
What's missing here (and was missing before!) is the _why_ of this, can you explain that in the comment?